File: 03/clojure/src/switch.clj
;; switch.clj
;; Author: Brian Goins (bmg23)
;; Date: 2/16/2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns switch)


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;Function letterGrade
;;Input: average grade, Integer
;;Output: letter grade, string
;;Uses case statement to return letter grade.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn letterGrade [average]
    (cond
        (= (int (/ average 10)) 10)           "A"
        (= (int (/ average 10)) 9)             "A"
        (= (int (/ average 10)) 8)             "B"
        (= (int (/ average 10)) 7)             "C"
        (= (int (/ average 10)) 6)             "D"
        :else                                  "F"      
    )
)

(defn -main []
    (print "\nEnter the average grade: ")
    (flush)
    (let
        [ avgGrade (read) ]

        (print "\nLetter Grade: ")
        (print (letterGrade avgGrade))
        (print "\n") (flush)
    
    )
)

File: 03/switch.rb
## switch.rb
## Author: Brian Goins (bmg23)
## Date: 2/16/2019
###############################


#############################################
## Function letterGrade
## Input: average grade, Integer
## Output: letter grade, string
## Uses case statement to return letter grade.
##############################################

def letterGrade(average) 
    case (average / 10)
        when 9..10
            return "A"
        when 8..9
            return "B"
        when 7..8
            return "C" 
        when 6..7
            return "D"
        else 
            return "F"
    end
end  

if __FILE__ == $0 
    print "\nEnter average grade: "
    avgGrade = gets.chomp.to_f
    print "\nLetter Grade: "
    puts letterGrade(avgGrade)
    print "\n"
end 